src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -82,6 +83,7 @@ public static Parameter FromInputParameter(in InputParameter operationParameter,
 
         private static Constant? GetDefaultValue(InputParameter operationParameter, TypeFactory typeFactory) => operationParameter switch
         {
+            { Name: "$host", DefaultValue.Value: ""} => (Constant?)null, // In M4, when endpoint is not set, its default value is an empty string instead of null
```

Comment:
should it be 
```suggestion
            { IsEndpoint: true, DefaultValue.Value: "" } => (Constant?)null, // In M4, when endpoint is not set, its default value is an empty string instead of null
```
and if in the modelerfour case, the corresponding parameter does not have `IsEndpoint: true` when it is actually endpoint (or host), we should fix there

Reply:
I can do this. But my concern is it is just a workaround for M4's bug (we might not want to fix M4), so I just want to limit this condition to swagger, not want to impact typespec. If we use `IsEndpoint`, typespec will be included to this change.

Reply:
fair enough

Reply:
If it is a workaround for swagger, it may be good to add this logic in CodeModelConvert where convert CodeModel to InputModel.

Reply:
> If it is a workaround for swagger, it may be good to add this logic in CodeModelConvert where convert CodeModel to InputModel.

well this is my point as well - but if there is difficulties doing this, I am fine with the current solution because in typespec I believe `$host` is never a valid identifier name therefore it should be fine.

Reply:
I would prefer to add workaround
1. M4 is in maintaining mode.
2. It's better to get this PR in early because it changes API.